blockchain/types/tx_internal_data_dynamic_fee.go

```diff
@@ -0,0 +1,512 @@
+	}
+
+	if v, ok := values[TxValueKeyTo].(common.Address); ok {
+		d.Recipient = &v
+		delete(values, TxValueKeyTo)
+	} else {
+		return nil, errValueKeyToMustAddress
+	}
+
+	if v, ok := values[TxValueKeyAmount].(*big.Int); ok {
+		d.Amount.Set(v)
+		delete(values, TxValueKeyAmount)
+	} else {
+		return nil, errValueKeyAmountMustBigInt
+	}
+
+	if v, ok := values[TxValueKeyData].([]byte); ok {
+		d.Payload = common.CopyBytes(v)
+		delete(values, TxValueKeyData)
+	} else {
+		return nil, errValueKeyDataMustByteSlice
+	}
+
+	if v, ok := values[TxValueKeyGasLimit].(uint64); ok {
+		d.GasLimit = v
+		delete(values, TxValueKeyGasLimit)
+	} else {
+		return nil, errValueKeyGasLimitMustUint64
+	}
+
+	if v, ok := values[TxValueKeyGasFeeCap].(*big.Int); ok {
+		d.GasFeeCap.Set(v)
+		delete(values, TxValueKeyGasFeeCap)
+	} else {
+		return nil, errValueKeyGasFeeCapMustBigInt
+	}
+	if v, ok := values[TxValueKeyGasTipCap].(*big.Int); ok {
+		d.GasTipCap.Set(v)
+		delete(values, TxValueKeyGasTipCap)
+	} else {
+		return nil, errValueKeyGasTipCapMustBigInt
+	}
+	if v, ok := values[TxValueKeyAccessList].(AccessList); ok {
+		d.AccessList = make(AccessList, len(v))
+		copy(d.AccessList, v)
+		delete(values, TxValueKeyAccessList)
+	} else {
+		return nil, errValueKeyAccessListInvalid
+	}
+
+	if v, ok := values[TxValueKeyChainID].(*big.Int); ok {
+		d.ChainID.Set(v)
+		delete(values, TxValueKeyChainID)
+	} else {
+		return nil, errValueKeyChainIDInvalid
+	}
+
+	if len(values) != 0 {
+		for k := range values {
+			logger.Warn("unnecessary key", k.String())
+		}
+		return nil, errUndefinedKeyRemains
+	}
+
+	return d, nil
+}
+
+func (t *TxInternalDataDynamicFee) Type() TxType {
+	return TxTypeDynamicFee
+}
+
+func (t *TxInternalDataDynamicFee) GetRoleTypeForValidation() accountkey.RoleType {
+	return accountkey.RoleTransaction
+}
+
+func (t *TxInternalDataDynamicFee) GetAccountNonce() uint64 {
+	return t.AccountNonce
+}
+
+func (t *TxInternalDataDynamicFee) GetPrice() *big.Int {
+	return t.GasFeeCap
+}
+
+func (t *TxInternalDataDynamicFee) GetGasLimit() uint64 {
+	return t.GasLimit
+}
+
+func (t *TxInternalDataDynamicFee) GetRecipient() *common.Address {
+	return t.Recipient
+}
+
+func (t *TxInternalDataDynamicFee) GetAmount() *big.Int {
+	return new(big.Int).Set(t.Amount)
+}
+
+func (t *TxInternalDataDynamicFee) GetHash() *common.Hash {
+	return t.Hash
+}
+
+func (t *TxInternalDataDynamicFee) GetPayload() []byte {
+	return t.Payload
+}
+
+func (t *TxInternalDataDynamicFee) GetAccessList() AccessList {
+	return t.AccessList
+}
+
+func (t *TxInternalDataDynamicFee) GetGasTipCap() *big.Int {
+	return t.GasTipCap
+}
+
+func (t *TxInternalDataDynamicFee) GetGasFeeCap() *big.Int {
+	return t.GasFeeCap
+}
+
+func (t *TxInternalDataDynamicFee) SetHash(hash *common.Hash) {
+	t.Hash = hash
+}
+
+func (t *TxInternalDataDynamicFee) SetSignature(signatures TxSignatures) {
+	if len(signatures) != 1 {
+		logger.Crit("AccessListTransaction can receive only single signature!")
```

Comment:
```suggestion
		logger.Crit("TxTypeDynamicFee can receive only single signature!")
```

Reply:
I modified log message, would you check it?